lib/decimal.ex

```diff
@@ -731,6 +731,33 @@ defmodule Decimal do
   end
 
   @doc """
+  Parses a string into a decimal.
+
+  If successful, returns a tuple in the form of `{:ok, decimal}.
+  Otherwise `:error`.
+
+  ## Examples
+
+      iex> Decimal.parse("3.14")
+      {:ok, Decimal.new(3.14)}
+
+      iex> Decimal.parse("-1.1e3")
+      {:ok, Decimal.new(-1.1e3)}
+
+      iex> Decimal.parse("bad")
+      :error
+  """
+  @spec parse(String.t) :: {:ok, t} | :error
+  def parse(binary) do
+    try do
+      {:ok, Decimal.new(binary)}
+    rescue
```

Comment:
this rescue is ugly here. I thought about changing `new/1` but:
- people might be depending on it raising an error
- it would be backwards incompatible

I think `new/1` **with** strings could be deprecated and eventually unsupported.